composeApp/src/commonMain/kotlin/data/db/DBHelper.kt

```diff
@@ -0,0 +1,61 @@
+package data.db
+
+import app.cash.sqldelight.ColumnAdapter
+import app.cash.sqldelight.db.SqlDriver
+import com.monstarlab.kmp.ProductDB
+import com.monstarlab.kmp.ProductDatabase
+import domain.model.Product
+
+class DBHelper(sqlDriver: SqlDriver) {
+    private val listOfStringsAdapter = object : ColumnAdapter<List<String>, String> {
+        override fun decode(databaseValue: String) =
+            if (databaseValue.isEmpty()) {
+                listOf()
+            } else {
+                databaseValue.split(",")
+            }
+
+        override fun encode(value: List<String>) = value.joinToString(separator = ",")
+    }
+
+    private val dbRef = ProductDatabase(sqlDriver, ProductDB.Adapter(listOfStringsAdapter))
+
+    fun insertProduct(product: Product) =
+        dbRef.productQueries.insertProduct(product.toDatabaseModel())
+
+    fun fetchAllProducts() =
```

Comment:
Should we introduce a mapper and then map it to the Domain model?

Reply:
Thinking about it as well.